apps/studio/schemas/boardYear.ts

```diff
@@ -0,0 +1,67 @@
+import { HeartIcon } from "@sanity/icons";
+
+const organizerSearch = [
+  {
+    type: "object",
+    fields: [
+      {
+        name: "organizer",
+        type: "reference",
+        title: "Organizer",
+        to: {
+          type: "organizer",
+        },
+        validation: (Rule) => Rule.required(),
+      },
+      {
+        name: "position",
+        type: "string",
+        title: "Position",
+        validation: (Rule) => Rule.required(),
+      },
+    ],
+    preview: {
+      select: {
+        title: "organizer.name",
+        subtitle: "position",
+      },
+    },
+    icon: HeartIcon,
+  },
+];
+
+const nameTitleAssoc: Map<string, string> = new Map([
+  ["corporate", "Corporate Organizers"],
+  ["logistics", "Logistics Organizers"],
+  ["marketing", "Marketing Organizers"],
+  ["tech", "Tech Organizers"],
+]);
+
+export default {
+  name: "boardYears",
+  type: "document",
+  title: "Board Years",
+  fields: [
+    {
+      name: "year",
+      type: "number",
+      title: "Academic Year",
+      validation: (Rule) => Rule.required(),
+    },
+    ...Array.from(nameTitleAssoc.entries()).map((l) => ({
+      name: l[0],
+      type: "array",
+      title: l[1],
+      of: organizerSearch,
+      validation: (Rule) => Rule.required(),
+    })),
```

Comment:
Personally, I'm not quite sure how I feel about this. Glancing over it feels a lot less readable than just listing out all four departments.
Either way I don't think the `Array.from()` is required here. `.map()` should already return an array of objects.

Reply:
Yeah, it was either this or copy paste of the internal dictionary, one for each department.

Reply:
Also in this, I needed to use `Array.from()` because `.entries()` returns an iterable, not an array and `.map` requires an array in order to work.

Reply:
I think this might be a little more readable:

```ts
const nameTitleAssoc = {
  corporate: "Corporate Organizers",
  logistics: "Logistics Organizers",
  marketing: "Marketing Organizers",
  tech: "Tech Organizers",
};
```
```ts
...Object.entries(nameTitleAssoc).map(([key, title]) => ({
  name: key,
  type: "array",
  title: title,
  of: organizerSearch,
  validation: (Rule) => Rule.required(),
})),
```

Reply:
That looks a lot more readable! I'll update it to this then :+1:

Reply:
Fixed in [`babb28d`](https://github.com/HackAtUCI/hack.ics.uci.edu/pull/12/commits/babb28de063249dfcad0d491a4467125cb60e11d)